src/oscar/core/application.py

```diff
@@ -1,10 +1,17 @@
 from django.urls import reverse_lazy
-from django.urls.resolvers import RegexURLPattern as URLPattern
 
 from oscar.core.loading import feature_hidden
 from oscar.views.decorators import permissions_required
 
 
+try:
+    # Django 2
+    from django.urls import URLPattern
+except ImportError:
+    # Django 1.11
+    from django.urls.resolvers import RegexURLPattern as URLPattern
```

Comment:
I believe we'd move that to `core.compat`.

Reply:
I'm not sure why? You'd end up with unused import warnings in `compat.py`.

Reply:
Right. Let's then leave it as is.